packages/langium/src/lsp/document-update-handler.ts

```diff
@@ -83,6 +89,23 @@ export class DefaultDocumentUpdateHandler implements DocumentUpdateHandler {
         });
     }
 
+    /**
+     * Check whether the service registry contains a service instance for the given URI.
+     *
+     * Some language clients (vscode) decide to send update notifications for files that have been renamed to a different file extension.
+     * In this case, the service registry may not contain a service for the new URI. We have to ignore any changes to those files.
+     *
+     * In case we only have a single language in our registry, we can safely use that language for all URIs.
+     */
+    protected hasService(uri: URI): boolean {
+        try {
+            this.serviceRegistry.getServices(uri);
+            return true;
+        } catch {
+            return false;
+        }
```

Comment:
Shall we introduce a `hasServices` method in ServiceRegistry? This try-catch works, but looks like a workaround for a missing service feature.

Reply:
Yeah, makes sense 👍